crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2287,6 +2293,50 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn does_not_duplicate_global_registration_2311() {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        let mqtt_schema = MqttSchema::new();
+        let measurement_topic = mqtt_schema.topic_for(
+            &"device/main/service/my_measurement_service"
+                .parse()
+                .unwrap(),
+            &tedge_api::mqtt_topics::Channel::Measurement {
+                measurement_type: "my_type".to_string(),
+            },
+        );
```

Comment:
In tests, feel free to use raw representation of the topics and payloads, rather than complicating it with `MqttSchema` and all. The readability would be much better if we can just see the topics and payloads directly, rather than assuming what the `MqttSchema` schema would have produced.

Reply:
In the future, I'd like to move away from using plain MQTT topics and contain them only to places which deal with MQTT directly, i.e. the MQTT mapper. I agree though that using `MqttSchema` is a bit of a pain right now, not only here, but everywhere, and it's partly because all the code has to be aware of the MQTT root in order to generate messages. I'll see if this can be simpified.